File: my-folder/0001-two-sum/solution.py


class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        numToIdx = {}

        for i, n in enumerate(nums):
            diff = target - n
            if diff in numToIdx:
                return [i, numToIdx.get(diff)]
            numToIdx[n] = i
        return []

File: my-folder/0001-two-sum/solution.ts
function twoSum(nums: number[], target: number): number[] {
  const numIdxMap = new Map<number, number>();

  for (let i = 0; i < nums.length; i++) {
    const diff = target - nums[i];
    if (numIdxMap.has(diff)) {
      return [i, numIdxMap.get(diff)!];
    }
    numIdxMap.set(nums[i], i);
  }

  return [];
}
